packages/utilities/src/utils/common.ts

```diff
@@ -27,54 +27,104 @@ export const extractUrlAndGlobal = (urlAndGlobal: string): [string, string] => {
 };
 
 const getRuntimeRemotes = () => {
-  //@ts-ignore
-  const remoteVars = (process.env.REMOTES || {}) as RemoteVars;
-
-  const runtimeRemotes = Object.entries(remoteVars).reduce(function (
-    acc,
-    item
-  ) {
-    const [key, value] = item;
-    // if its an object with a thenable (eagerly executing function)
-    if (typeof value === 'object' && typeof value.then === 'function') {
-      acc[key] = { asyncContainer: value };
-    }
-    // if its a function that must be called (lazily executing function)
-    else if (typeof value === 'function') {
-      // @ts-ignore
-      acc[key] = { asyncContainer: value };
-    }
-    // if its just a string (global@url)
-    else if (typeof value === 'string') {
-      const [url, global] = extractUrlAndGlobal(value);
-      acc[key] = { global, url };
-    }
-    // we dont know or currently support this type
-    else {
-      //@ts-ignore
-      console.log('remotes process', process.env.REMOTES);
-      throw new Error(
-        `[mf] Invalid value received for runtime_remote "${key}"`
-      );
-    }
-    return acc;
-  },
-  {} as RuntimeRemotesMap);
+  try {
+    //@ts-ignore
+    const remoteVars = (process.env.REMOTES || {}) as RemoteVars;
+
+    const runtimeRemotes = Object.entries(remoteVars).reduce(function (
+      acc,
+      item
+    ) {
+      const [key, value] = item;
+      // if its an object with a thenable (eagerly executing function)
+      if (typeof value === 'object' && typeof value.then === 'function') {
+        acc[key] = { asyncContainer: value };
+      }
+      // if its a function that must be called (lazily executing function)
+      else if (typeof value === 'function') {
+        // @ts-ignore
+        acc[key] = { asyncContainer: value };
+      }
+      // if its just a string (global@url)
+      else if (typeof value === 'string') {
+        const [url, global] = extractUrlAndGlobal(value);
+        acc[key] = { global, url };
+      }
+      // we dont know or currently support this type
+      else {
+        //@ts-ignore
+        console.log('remotes process', process.env.REMOTES);
+        throw new Error(
```

Comment:
if there is couple of remotes failed to load, wont you prefer notify the user about all of the problematic remotes?
here, it will fail when the first one is failed to process

Reply:
good point, something that can be added to the backlog as i think id need to loop over everything then start constructing the error stack